File: app/src/main/java/com/example/oopsiecard/Repository.java
package com.example.oopsiecard;

import android.app.Application;

import java.util.List;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class Repository {

    private PunchedHolesDAO mPunchedHolesDAO;
    private PunchedHoles mPunchedHoles;
    private List<PunchedHoles> allPunchedHoles;

    private static final int NUMBER_OF_THREADS = 4;

    static final ExecutorService databaseExecutor = Executors.newFixedThreadPool(NUMBER_OF_THREADS);

    public Repository(Application application){
        DatabaseBuilder db = DatabaseBuilder.getDatabase(application.getApplicationContext());

        assert db != null;
        mPunchedHolesDAO = db.punchedHolesDAO();
    }




}

File: app/src/main/java/com/example/oopsiecard/Repository.kt
package com.example.oopsiecard

import android.app.Application
import com.example.oopsiecard.DatabaseBuilder.Companion.getDatabase
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

class Repository(application: Application) {
    private val mPunchedHolesDAO: PunchedHolesDAO
    private val mPunchedHoles: PunchedHoles? = null

    var allPunchedHoles: List<Int>? = null
    private val NUMBER_OF_THREADS = 4
    private val databaseExecutor: ExecutorService = Executors.newFixedThreadPool(NUMBER_OF_THREADS)

    init {
        val db = getDatabase(application.applicationContext)!!
        mPunchedHolesDAO = db.punchedHolesDAO()
    }

    fun insert(punchedHoles: PunchedHoles?) {
        databaseExecutor.execute { mPunchedHolesDAO.insert(punchedHoles) }
    }

    fun update(punchedHoles: PunchedHoles?) {
        databaseExecutor.execute { mPunchedHolesDAO.update(punchedHoles) }
    }

    fun delete(punchedHoles: PunchedHoles?) {
        databaseExecutor.execute { mPunchedHolesDAO.delete(punchedHoles) }
    }

    fun allHoles(){
        databaseExecutor.execute {allPunchedHoles = mPunchedHolesDAO.allHoles()}

    }

}